packages/static-cs-lite/lib/image/createImage.js

```diff
@@ -0,0 +1,155 @@
+    }
+
+    // convert color space
+    if (isColorImage) {
+      const context = canvas.getContext("2d");
+      const imageData = context.createImageData(imageFrame.columns, imageFrame.rows);
+
+      // imageData.data is being changed by reference.
+      imageFrameUtils.convert.colorSpace(imageFrame, imageData.data);
+      if (!imageData.data) {
+        throw new Error("Missing image data after converting color space");
+      }
+      imageFrame.imageData = imageData;
+      imageFrame.pixelData = imageData.data;
+    }
+  }
+
+  if ((!imageFrame.smallestPixelValue || !imageFrame.largestPixelValue || imageFrame.pixelData.constructor, originalDataConstructor)) {
+    // calculate smallest and largest PixelValue of the converted pixelData
+    const { min, max } = imageFrameUtils.get.pixelDataMinMax(imageFrame.pixelData);
+
+    imageFrame.smallestPixelValue = min;
+    imageFrame.largestPixelValue = max;
+  }
+
+  const image = {
+    color: isColorImage,
+    columnPixelSpacing,
+    columns: imageFrame.columns,
+    height: imageFrame.rows,
+    preScale: imageFrame.preScale,
+    intercept,
+    slope,
+    invert: imageFrame.photometricInterpretation === "MONOCHROME1",
+    minPixelValue: imageFrame.smallestPixelValue,
+    maxPixelValue: imageFrame.largestPixelValue,
+    rowPixelSpacing,
+    rows: imageFrame.rows,
+    sizeInBytes: imageFrame.pixelData.byteLength,
+    width: imageFrame.columns,
+    windowCenter,
+    windowWidth,
+    decodeTimeInMS: imageFrame.decodeTimeInMS,
+    floatPixelData: undefined,
+    imageFrame,
+  };
+
+  // If pixel data is intrinsically floating 32 array, we convert it to int for
+  // display in cornerstone. For other cases when pixel data is typed as
+  // Float32Array for scaling; this conversion is not needed.
+  if (imageFrame.pixelData instanceof Float32Array && convertFloatPixelDataToInt) {
+    const floatPixelData = imageFrame.pixelData;
+    const results = imageFrameUtils.get.pixelDataIntType(floatPixelData);
+
+    image.minPixelValue = results.min;
+    image.maxPixelValue = results.max;
+    image.slope = results.slope;
+    image.intercept = results.intercept;
+    image.floatPixelData = floatPixelData;
+    image.getPixelData = () => results.intPixelData;
+  } else {
+    image.getPixelData = () => imageFrame.pixelData;
+  }
+
+  if (image.color) {
+    // let lastImageIdDrawn;
+    image.getCanvas = () => {
+      canvas.height = image.rows;
+      canvas.width = image.columns;
+      const context = canvas.getContext("2d");
+
+      context.putImageData(imageFrame.imageData, 0, 0);
+
+      return canvas;
+    };
+  }
+
+  // Modality LUT
+  if (modalityLUTSequence && modalityLUTSequence.length > 0 && imageFrameUtils.is.modalityLUT(sopClassUID)) {
+    image.modalityLUT = modalityLUTSequence[0];
+  }
+
+  // VOI LUT
+  if (voiLUTSequence && voiLUTSequence.length > 0) {
+    image.voiLUT = voiLUTSequence[0];
+  }
+
+  if (image.color) {
+    image.windowWidth = 255;
```

Comment:
These values should be 256 and 128, according to the comment in DICOM part 3:
A Window Center of 2^(n-1) and a Window Width of 2^n selects the range of input values from 0 to 2n-1. This represents a mathematical identity VOI LUT transformation over the possible input values (whether used or not) in the case where no Modality LUT is specified and the stored pixel data are n bit unsigned integers.

Reply:
ok, thank you for the input